testsuite/cli/src/client_proxy.rs

```diff
@@ -172,12 +172,15 @@ impl ClientProxy {
             Some(dd_account_data)
         };
 
-        let faucet_server = match faucet_server {
-            Some(server) => server,
-            None => url
-                .host_str()
-                .ok_or_else(|| format_err!("Missing host in URL"))?
-                .replace("client", "faucet"),
+        let faucet_url = match faucet_url {
```

Comment:
```
facuet_url.map_or( Url::parse(format!(...)),  |x| Url::parse(x)).ok_or_else (|| fomrat_err!(""))?; 
```

Reply:
Couldn't figure this out exactly, did something slightly different with `unwrap_or()`.